hack/snapshot_to_catalog.sh

```diff
@@ -0,0 +1,172 @@
+  exit 1
+fi
+
+# temporary file for rendering the bundle part of the catalog
+TEMP_BUNDLE_FILE=$(mktemp)
+
+cleanup() {
+  # remove temporary snapshot file
+  if [ -n "${TMP_SNAPSHOT_JSON}" ]; then
+    rm -f "${TMP_SNAPSHOT_JSON}"
+  fi
+
+  # remove temporary bundle file
+  if [ -n "${TEMP_BUNDLE_FILE}" ]; then
+    rm -f "${TEMP_BUNDLE_FILE}"
+  fi
+
+}
+
+trap cleanup EXIT
+
+: ${OPM:=$(command -v opm)}
+echo "using opm from ${OPM}"
+# check if opm version is v1.39.0 or exit
+if ! ${OPM} version | grep -q "v1.27.1"; then
+  echo "opm version v1.27.1 is required"
+  exit 1
+fi
+
+: ${YQ:=$(command -v yq)}
+echo "using yq from ${YQ}"
+# check if yq exists
+if [ -z "${YQ}" ]; then
+  echo "yq is required"
+  exit 1
+fi
+
+BUNDLE_IMAGE_BASE="registry.redhat.io/openshift-lightspeed-beta/lightspeed-operator-bundle"
+OPERATOR_IMAGE_BASE="registry.redhat.io/openshift-lightspeed-beta/lightspeed-rhel9-operator"
+CONSOLE_IMAGE_BASE="registry.redhat.io/openshift-lightspeed-beta/lightspeed-console-plugin-rhel9"
+SERVICE_IMAGE_BASE="registry.redhat.io/openshift-lightspeed-beta/lightspeed-service-api-rhel9"
+
+BUNDLE_IMAGE_ORIGIN=$(jq -r '.spec.components[]| select(.name=="bundle") | .containerImage' "${TMP_SNAPSHOT_JSON}")
+BUNDLE_IMAGE=$(sed 's|quay\.io/redhat-user-workloads/crt-nshift-lightspeed-tenant/ols/bundle|'"${BUNDLE_IMAGE_BASE}"'|g' <<<${BUNDLE_IMAGE_ORIGIN})
+
+OPERATOR_IMAGE=$(jq -r '.spec.components[]| select(.name=="lightspeed-operator") | .containerImage' "${TMP_SNAPSHOT_JSON}")
+OPERATOR_IMAGE=$(sed 's|quay\.io/redhat-user-workloads/crt-nshift-lightspeed-tenant/ols/lightspeed-operator|'"${OPERATOR_IMAGE_BASE}"'|g' <<<${OPERATOR_IMAGE})
+
+CONSOLE_IMAGE=$(jq -r '.spec.components[]| select(.name=="lightspeed-console") | .containerImage' "${TMP_SNAPSHOT_JSON}")
+CONSOLE_IMAGE=$(sed 's|quay\.io/redhat-user-workloads/crt-nshift-lightspeed-tenant/ols/lightspeed-console|'"${CONSOLE_IMAGE_BASE}"'|g' <<<${CONSOLE_IMAGE})
+
+SERVICE_IMAGE=$(jq -r '.spec.components[]| select(.name=="lightspeed-service") | .containerImage' "${TMP_SNAPSHOT_JSON}")
+SERVICE_IMAGE=$(sed 's|quay\.io/redhat-user-workloads/crt-nshift-lightspeed-tenant/ols/lightspeed-service|'"${SERVICE_IMAGE_BASE}"'|g' <<<${SERVICE_IMAGE})
+
+RELATED_IMAGES=$(
+  cat <<-EOF
+[
+  {
+    "name": "lightspeed-service-api",
+    "image": "${SERVICE_IMAGE}"
+  },
+  {
+    "name": "lightspeed-console-plugin",
+    "image": "${CONSOLE_IMAGE}"
+  },
+  {
+    "name": "lightspeed-operator",
+    "image": "${OPERATOR_IMAGE}"
+  },
+  { "name": "lightspeed-operator-bundle",
+    "image": "${BUNDLE_IMAGE}"
+  }
+]
+EOF
+)
+
+echo "Catalog will use the following images:"
+echo "BUNDLE_IMAGE=${BUNDLE_IMAGE}"
+echo "OPERATOR_IMAGE=${OPERATOR_IMAGE}"
+echo "CONSOLE_IMAGE=${CONSOLE_IMAGE}"
+echo "SERVICE_IMAGE=${SERVICE_IMAGE}"
+
+echo BUNDLE_IMAGE_ORIGIN=${BUNDLE_IMAGE_ORIGIN}
+${OPM} render ${BUNDLE_IMAGE_ORIGIN} --output=yaml >"${TEMP_BUNDLE_FILE}"
```

Comment:
@raptorsun TEMP_BUNDLE_FILE is this the right name , the render output is catalog config right ? I am baffled by the naming . if not , this PR LGTM

Reply:
That TEMP_BUNDLE_FILE is  the bundle part of the catalog  that the `opm` renders. It is later assembled into the catalog file. Is a name like `TEMP_BUNDLE_PART_OF_CATALOG` better?

Reply:
Lets leave it as is  TEMP_BUNDLE_FILE confused me .